src/safeds/data/tabular/transformation/_k_nearest_neighbors_imputer.py

```diff
@@ -108,6 +110,10 @@ def fit(self, table: Table) -> KNearestNeighborsImputer:
         else:
             column_names = self._column_names
             _check_columns_exist(table, column_names)
+        
+        if self._value_to_replace is None:
+            from numpy import nan
+            self._value_to_replace = nan
```

Comment:
You need a new local variable, rather than assigning to the field. Otherwise, this becomes visible when accessing `value_to_replace` after fitting.